### PR TITLE
fix: throw an exception if schema not found from SR during INSERT #9457

### LINKED ISSUE #9457: Avoid auto-registering key schema during INSERT if schema not found from Schema Registry
**Describe the bug**
During INSERT if the key schema is not found from Schema Registry, validation is skipped and it auto-registers a new schema.
This can confuse the user if the user has already created a schema for the underlying topic and the Schema Registry fails to respond possibly due to a Network Issue.

**To Reproduce**
https://github.com/confluentinc/ksql/blob/master/ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/InsertValuesExecutor.java#L418

**Expected behavior**
Throw an exception and fail the INSERT statement

**Actual behavior**
Auto registers a new key schema

### PR DESCRIPTION
**Pull request by bvarghese1**
### Description 
Fixes #9457 
Throws an exception and fails the insert to avoid auto-registering a new key schema when the schema is not found from Schema Registry

### Testing done 
Added a unit test

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Review by vpapavas — COMMENTED** [review]
Thank you @bvarghese1 ! I lack general context but this seems to achieve what you stated in the description. It would be nice to try it out locally with the CLI for example but we would need to partition off SR from ksql. WDYT?

**Comment by bvarghese1** [comment]
@vpapavas If I stop schema-registry and try an insert I get a different error message:
```
ksql> insert into sr_stream (ROWKEY, v1) values (struct(K1:=2, K2:=2.0), 2);
Failed to insert values into 'SR_STREAM'. Could not determine that insert values operations is safe; operation potentially overrides existing key schema in schema registry.
```

**Comment by bvarghese1** [comment]
If I go and manually delete the schema from Schema Registry and then try to insert it hits the condition that this PR fixes:
```
ksql> insert into sr_stream (ROWKEY, v1) values (struct(K1:=2, K2:=2.0), 2);
Failed to insert values into 'SR_STREAM'. Failed to fetch schema. Please check if schema exists in Schema Registry and/or check connection with Schema Registry.
```

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/InsertValuesExecutor.java [hunk 1]
```diff
@@ -418,10 +418,8 @@ private static Optional<Integer> ensureKeySchemasMatch(
     }
 
     if (!latest.isPresent()) {
-      // Note: This will result in auto-registering a key schema if the schema is not found from SR
-      // This is a known issue and, we will tackle this in a separate issue once we have determined
-      // whether we want to continue auto-registering during INSERT
-      return Optional.empty();
+      throw new KsqlException("Failed to fetch schema. Please check if schema exists in "
```
Comment (spena) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/InsertValuesExecutor.java:421]: Can you include the subject name the user can use to check? i.e. would the user use `topic-key` or `topic-value`?

Could you also specify `Failed to fetch the key schema.`?

Comment (bvarghese1) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/InsertValuesExecutor.java]: Done

**Comment by vpapavas** [comment]
Awesome! Thank you for trying that out. LGTM!

**Review by vpapavas — APPROVED** [review]

**Review by spena — APPROVED** [review]
Looks good. Can we keep this on hold until we get green light from @colinhicks that we don't want to auto register the key schema if it is not found?

**Review by colinhicks — APPROVED** [review]
LGTM. Thanks @bvarghese1!

**Pull request merged**